In [2]:
import pandas as pd
import numpy as np

In [13]:
ARQUIVO= '01.amazon_sales_dataset.csv'
df = pd.read_csv(ARQUIVO)
print(df.head())

coluna_desconto = df['discount']

desconto_medio = coluna_desconto.mean()

desconto_mediana = coluna_desconto.median()

print(f"Desconto Médio (Média): {desconto_medio}")
print(f"Valor Central (Mediana): {desconto_mediana}")


  order_id  order_date   ship_date delivery_date order_status customer_id  \
0   A10000  2026-01-31  2026-01-31    2026-01-08    Delivered       C5691   
1   A10001  2026-01-20  2026-02-03    2026-02-03    Delivered       C9811   
2   A10002  2026-01-15  2026-02-07    2026-01-03    Delivered       C7341   
3   A10003  2026-01-18  2026-01-15    2026-01-20    Delivered       C4012   
4   A10004  2026-01-27  2026-01-04    2026-01-23    Delivered       C1328   

        customer_name country           state                 city  ...  \
0        Ricky Potter   India  South Carolina              New Joe  ...   
1     Chris Davenport   India       Tennessee         Madisonmouth  ...   
2   Timothy Gallagher   India            Iowa       East Larryberg  ...   
3      Angela Collins   India        Kentucky  South Margaretshire  ...   
4  David Davidson DDS   India    North Dakota        Velasquezview  ...   

  product_name     category sub_category                        brand  \
0      withou

In [14]:
coluna_desconto = df['discount']

desconto_minimo = coluna_desconto.min()
desconto_maximo = coluna_desconto.max()

q1 = coluna_desconto.quantile(0.25)
q3 = coluna_desconto.quantile(0.75)

amplitude = desconto_maximo - desconto_minimo

print(f"Mínimo (Menor desconto): {desconto_minimo}")
print(f"Máximo (Maior desconto): {desconto_maximo}")
print(f"Q1 (Primeiro Quartil / 25%): {q1}")
print(f"Q3 (Terceiro Quartil / 75%): {q3}")
print(f"Amplitude Total: {amplitude}")

Mínimo (Menor desconto): 0.0
Máximo (Maior desconto): 0.3
Q1 (Primeiro Quartil / 25%): 0.07
Q3 (Terceiro Quartil / 75%): 0.22
Amplitude Total: 0.3


In [15]:
coluna_desconto = df['discount']

assimetria = coluna_desconto.skew()

print(f"Assimetria dos dados de desconto: {assimetria}")

Assimetria dos dados de desconto: -0.004164539978905712


In [ ]:
coluna_desconto = df['discount']

menor_desconto = coluna_desconto.min()
maior_desconto = coluna_desconto.max()

media = coluna_desconto.mean()
mediana = coluna_desconto.median()

diferenca = media - mediana

razao = media / mediana

assimetria = coluna_desconto.skew()

print("=== TAREFA 1: LIMITES DOS DADOS ===")
print(f"Menor Desconto: {menor_desconto}")
print(f"Maior Desconto: {maior_desconto}\n")

print("=== TAREFA 2: AVALIAÇÃO DO DELTA ===")
print(f"Média: {media}")
print(f"Mediana: {mediana}")
print(f"Diferença (Média - Mediana): {diferenca}")
print(f"Razão (Média / Mediana): {razao:.4f}")
print(f"Assimetria (Skewness): {assimetria:.4f}")

=== TAREFA 1: LIMITES DOS DADOS ===
Menor Desconto: 0.0
Maior Desconto: 0.3

=== TAREFA 2: AVALIAÇÃO DO DELTA ===
Média: 0.14996800000000002
Mediana: 0.15
Diferença (Média - Mediana): -3.199999999997649e-05
Razão (Média / Mediana): 0.9998
Assimetria (Skewness): -0.0042


                          RELATÓRIO DESCRITIVO

1. Introdução
Este relatório apresenta uma análise estatística descritiva detalhada acerca dos descontos aplicados
nos registros da base de dados carregada. O objetivo fundamental desta análise é compreender o
comportamento da política de descontos, mapear os limites operacionais (valores mínimos e máximos),
identificar as medidas de tendência central e de posição, e avaliar o equilíbrio e a assimetria da
distribuição dos dados.

2. Metodologia Computacional
A análise foi desenvolvida em ambiente computacional utilizando o editor Visual Studio Code
(VSCode) e a linguagem de programação Python. Para a manipulação dos dados estruturados e
computação das métricas estatísticas, utilizou-se a biblioteca Pandas, aplicando-se métodos nativos
para garantia de precisão e performance.

3. Análise Descritiva e Medidas de Posição
As medidas de posição fornecem uma visão estruturada sobre como os descontos estão distribuídos ao
longo de intervalos específicos. A tabela abaixo resume as métricas computadas no script:
Métrica Estatística / Função Pandas / Valor Obtido
Mínimo (Menor Desconto) / .min() / 0.0
Primeiro Quartil (Q1 - 25%) / .quantile(0.25) / 0.07
Mediana (Q2 - 50%) / .median() / 0.15
Terceiro Quartil (Q3 - 75%) / .quantile(0.75) / 0.22
Máximo (Maior Desconto) / .max() / 0.3
Amplitude Total / Máx - Mín / 0.3

O intervalo entre o Primeiro Quartil (Q1) e o Terceiro Quartil (Q3) representa o Intervalo Interquartílico
(IQR), onde estão concentrados os 50% centrais dos descontos aplicados, isolando valores extremos de
cauda.

4. Medidas de Tendência Central e Avaliação do Delta
Para avaliar o ponto de equilíbrio dos dados, computou-se a Média ( .mean() ) e a Mediana
( .median() ). A relação de distanciamento entre essas duas métricas é quantificada por meio do Delta
(Diferença Absoluta) e da Razão Proporcional:

Diferença = Média − Mediana
Razão = Média ÷ Mediana

Interpretação do Delta e Equilíbrio:
Média: 0.14996800000000002 | Mediana: 0.15
Diferença Absoluta: -3.199999999997649e-05
Razão Proporcional: 0.9998

5. Análise de Assimetria (Skewness)
O cálculo da assimetria através do método .skew() resultou no coeficiente de -0.0042 . Esta métrica categoriza matematicamente a distribuição da seguinte forma:

• Distribuição Aproximadamente Simétrica (Coeficiente entre -0.5 e 0.5): Indica que a política de
descontos está distribuída de maneira homogênea e equilibrada ao redor do centro. A quantidade e
intensidade de descontos altos e baixos se equivalem.
• Distribuição Assimétrica Positiva / À Direita (Coeficiente > 0.5): Indica desequilíbrio. A cauda
longa está à direita, significando que a grande maioria dos registros possui descontos baixos, e a
média foi puxada para cima devido a poucos registros com descontos muito elevados.
• Distribuição Assimétrica Negativa / À Esquerda (Coeficiente < -0.5): Indica desequilíbrio. A
cauda longa está à esquerda, denotando que o volume de dados concentra-se em descontos altos,
existindo apenas casos pontuais e isolados com penalidades ou descontos ínfimos.

6. Conclusão
Com base nas métricas extraídas, conclui-se que o comportamento da distribuição da variável de
desconto reflete uma estratégia padronizada.
Recomenda-se a utilização destes parâmetros estáticos para a modelagem de futuras margens e
previsões comerciais, atentando-se para o impacto que os valores extremos causam sobre a média real
de mercado.